src/epics/handleEmailLogin.js

```diff
@@ -49,16 +51,47 @@ const emailChangeVerifyGQL = gql`
   }
 `
 
+const NEWSLETTER_SUBSCRIPTION_MUTATION = gql`
+  mutation changeNewsletterSubscription(
+    $subscription: NewsletterSubscriptionType
+  ) {
+    changeNewsletterSubscription(newsletterSubscription: $subscription) {
+      newsletterSubscription
+    }
+  }
+`
+
 export const handleLoginSuccess = (action$, store, { client }) =>
   action$
     .ofType(actions.USER_LOGIN_SUCCESS)
     .mergeMap(action => {
       const { token, consent } = action
+
+      const hasSubscribed = localStorage.getItem(SUBSCRIPTION_FLAG)
+
+      if (hasSubscribed) {
+        localStorage.removeItem(SUBSCRIPTION_FLAG)
+      }
+
       return Observable.merge(
         Observable.of(showNotification('You are logged in!')),
+
         consent
           ? Observable.of(replace(`/consent?consent=${consent}&token=${token}`))
           : Observable.empty()
+      ).map(() =>
+        // NOTE(@vanguard): Delaying mutation because there is possible bug
```

Comment:
Ideally, you can wait for action about a user is logged in. But add TODO here now.